File: PHP/Endpoint Examples/Multipart Payload/merged-pdf.php
<?php
require 'vendor/autoload.php'; // Require the autoload file to load Guzzle HTTP client.

use GuzzleHttp\Client; // Import the Guzzle HTTP client namespace.
use GuzzleHttp\Psr7\Request; // Import the PSR-7 Request class.
use GuzzleHttp\Psr7\Utils; // Import the PSR-7 Utils class for working with streams.

$client = new Client(); // Create a new instance of the Guzzle HTTP client.

$headers = [
  'Api-Key' => 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' // Set the API key in the headers for authentication.
];

$options = [
  'multipart' => [
    [
      'name' => 'file', // Specify the field name for the first file.
      'contents' => Utils::tryFopen('/path/to/file', 'r'), // Open the file specified by the '/path/to/file' for reading.
      'filename' => '/path/to/file', // Set the filename for the first file to be merged, in this case, '/path/to/file'.
      'headers' => [
        'Content-Type' => '<Content-type header>' // Set the Content-Type header for the first file.
      ]
    ],
    [
      'name' => 'pages[]', // Specify the field name for the pages option for the first file.
      'contents' => '1-last' // Set the pages value for the first file (in this case, '1-last').
    ],
    [
      'name' => 'type[]', // Specify the field name for the type option for the first file.
      'contents' => 'file' // Set the type value for the first file (in this case, 'file').
    ],
    [
      'name' => 'file', // Specify the field name for the second file.
      'contents' => Utils::tryFopen('/path/to/file', 'r'), // Open the file specified by the '/path/to/file' for reading.
      'filename' => '/path/to/file', // Set the filename for the second file to be merged, in this case, '/path/to/file'.
      'headers' => [
        'Content-Type' => '<Content-type header>' // Set the Content-Type header for the second file.
      ]
    ],
    [
      'name' => 'pages[]', // Specify the field name for the pages option for the second file.
      'contents' => '1-last' // Set the pages value for the second file (in this case, '1-last').
    ],
    [
      'name' => 'type[]', // Specify the field name for the type option for the second file.
      'contents' => 'file' // Set the type value for the second file (in this case, 'file').
    ],
    [
      'name' => 'id[]', // Specify the field name for the ID option.
      'contents' => 'xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' // Set the value for the ID option (in this case, 'xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx').
    ],
    [
      'name' => 'pages[]', // Specify the field name for the pages option for the third file (identified by ID).
      'contents' => '1-last' // Set the pages value for the third file (identified by ID) (in this case, '1-last').
    ],
    [
      'name' => 'type[]', // Specify the field name for the type option for the third file (identified by ID).
      'contents' => 'id' // Set the type value for the third file (identified by ID) (in this case, 'id').
    ],
    [
      'name' => 'output', // Specify the field name for the output option.
      'contents' => 'pdfrest_merged_pdf' // Set the value for the output option (in this case, 'pdfrest_merged_pdf').
    ]
  ]
];

$request = new Request('POST', 'https://api.pdfrest.com/merged-pdf', $headers); // Create a new HTTP POST request with the API endpoint and headers.

$res = $client->sendAsync($request, $options)->wait(); // Send the asynchronous request and wait for the response.

echo $res->getBody(); // Output the response body, which contains the merged PDF content.

File: Python/Endpoint Examples/Multipart Payload/merged-pdf.py
from requests_toolbelt import MultipartEncoder
import requests
import json

merged_pdf_endpoint_url = 'https://api.pdfrest.com/merged-pdf'

# The /merged-pdf endpoint can take one or more PDF files or ids as input.
# This sample takes 2 PDF files and merges all the pages in the document into a single document.

merge_request_data = []

# Array of tuples that contains information about the 2 files that will be merged
files = [
    ('file_name.pdf', open('/path/to/file', 'rb'), 'application/pdf'),
    ('file_name2.pdf', open('/path/to/file', 'rb'), 'application/pdf')
]

# Structure the data that will be sent to POST merge request as an array of tuples
for i in range(len(files)):
    merge_request_data.append(("file", files[i]))
    merge_request_data.append(("pages", "1-last"))
    merge_request_data.append(("type", "file"))

merge_request_data.append(('output', 'example_mergedPdf_out'))

mp_encoder_mergedPdf = MultipartEncoder(
    fields=merge_request_data
)

# Let's set the headers that the merged-pdf endpoint expects.
# Since MultipartEncoder is used, the 'Content-Type' header gets set to 'multipart/form-data' via the content_type attribute below.
headers = {
    'Accept': 'application/json',
    'Content-Type': mp_encoder_mergedPdf.content_type,
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' # place your api key here
}

print("Sending POST request to merged-pdf endpoint...")
response = requests.post(merged_pdf_endpoint_url, data=mp_encoder_mergedPdf, headers=headers)

print("Response status code: " + str(response.status_code))

if response.ok:
    response_json = response.json()
    print(json.dumps(response_json, indent = 2))
else:
    print(response.text)

# If you would like to download the file instead of getting the JSON response, please see the 'get-resource-id-endpoint.py' sample.
